PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/cosmosspark/CosmosSparkClusterOpsCtrl.java

```diff
@@ -182,6 +200,95 @@ public CosmosSparkClusterOpsCtrl(@NotNull CosmosSparkClusterOps sparkServerlessC
                     } catch (Exception ex) {
                         log().error(ex.getMessage());
                     }
-                }, ex -> log().error(ex.getMessage(), ex));
+                }, ex -> log().error(ExceptionUtils.getStackTrace(ex)));
+
+        this.sparkServerlessClusterOps.getViewServerlessJobsAction()
+                .observeOn(ideSchedulers.dispatchUIThread())
+                .flatMap(accountNodePair -> {
+                    log().info(String.format("View serverless jobs message received. account: %s, node: %s",
+                            accountNodePair.getLeft().getName(), accountNodePair.getRight().getName()));
+                    // check if the requested job list tab exists in tool window
+                    AzureSparkServerlessAccount account = accountNodePair.getLeft();
+                    CosmosSparkADLAccountNode node = accountNodePair.getRight();
+                    ToolWindow toolWindow = ToolWindowManager.getInstance((Project) node.getProject()).getToolWindow("Cosmos Serverless Spark Jobs");
+                    Content existingContent= toolWindow.getContentManager().findContent(getDisplayName(account.getName()));
+                    if (existingContent != null) {
+                        // if the requested job list tab already exists in tool window,
+                        // show the existing job list tab
+                        toolWindow.getContentManager().setSelectedContent(existingContent);
+                        toolWindow.activate(null);
+                        return null;
```

Comment:
I support return null in flatMap would break it.

```suggestion
                        return Observable.empty();
```